qaframework-bdd-tests/pom.xml

```diff
@@ -62,7 +62,7 @@
 		<dependency>
 			<groupId>com.fasterxml.jackson.core</groupId>
 			<artifactId>jackson-databind</artifactId>
-			<version>2.11.2</version>
```

Comment:
oh we can merge them after all as long as they are compatible with our tests. cc @dkayiwa  @kdaud

Reply:
Or if they are not compatible but you have the time to fix the tests.

Reply:
> if they are not compatible but you have the time to fix the tests.

sure.

Reply:
My take on PRs about dependency upgrade is: We upgrade need to be tested first whether it's compatible with the tests than just merging. Let the changes be tested locally and if it none of the test is failing as a result of the upgrade then the merge is worth it.

Reply:
> Let the changes be tested locally and if it none of the test is failing as a result of the upgrade then the merge is worth it.

Doesn't the build automatically run tests and show you if they are failing or passing?

Reply:
@kdaud Let me know if there are any issues residing from this commit . otherwise like @dkayiwa  said, the pull request gets build automatically.

Reply:
> Doesn't the pull request build automatically run tests and show you if they are failing or passing?

No!

They only run on a PR if the change is located in their directory "qaframework-bdd-tests/src/**/*.*" for RefApp 2.x tests and - "qaframework-bdd-tests/cypress/**/*.*" for 3.x tests. Dependencies are residing residing in [pom.xml](https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/qaframework-bdd-tests/pom.xml) which does not lie in any of those path and tests won't run on the such PR that is why even none of the test ran on this PR. Additionally all the Platform UI tests only run on a push.

All said, we will only know the tests run result on a push that is why I suggested testing these changes before merging them.

Reply:
What prevents us from running those tests right from the pull request?

Reply:
It's due to the commit at https://github.com/openmrs/openmrs-contrib-qaframework/commit/9c2ccca019b94220b57d412b72e36eef08acb6ba

Reply:
Can't we fix that?

Reply:
Yes we can fix the issue but Platform UI tests won't be catered for as a result of their original set up to only run on a push.

Reply:
@kdaud but i think that is not related with jackson-databind upgrade ?

Reply:
Platform ui tests deal with changes to the platform. And this library upgrade is not in the platform.

Reply:
> @kdaud but i think that is not related with jackson-databind upgrade ?

Yes!

But have you followed the conversion from it's genesis?

Reply:
Was @dkayiwa  meaning to fix [this](https://github.com/openmrs/openmrs-contrib-qaframework/commit/9c2ccca019b94220b57d412b72e36eef08acb6ba) or fixing upgrades from dependant bot. How does openmrs-core handle dependant bot upgrades specifically.

Reply:
What I got from @dkayiwa is to make sure these pull requests from dependant bot build automatically run tests and show if they are failing or passing!

Reply:
I put up a PR at https://github.com/openmrs/openmrs-contrib-qaframework/pull/404 to fix the issue.